ClamavPlugin.inc.php

```diff
@@ -323,44 +327,63 @@ function clamscanHandleUpload($hookName, $args) {
 		//scan for viruses if file exists
 		if (null !== $file) {
 			$useSocket = $this->getSetting(CONTEXT_SITE, 'clamavUseSocket');
+			//Daemon
 			if ($useSocket === true) {
-				$message = $this->_clamDaemonFile($uploadedFile);
+				try {
+					$message = $this->_clamDaemonFile($uploadedFile);
+					//safe, continue with submission
+					if ($message == false) {
+						return false;
+					}
+				}
+				catch (ClamScanFailureException $e){
+					//couldn't scan, but ClamAV settings are permissive, continue with submission
+					if ( $this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')==="UNSCANNED_ALLOW") {
```

Comment:
To reference a constant, it should look like:
`$this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')===UNSCANNED_ALLOW`

This constant is currently only defined in the settings form, however, so it will need to be either moved to the Plugin class, or referenced from a settings form instance, like `=== $form::UNSCANNED_ALLOW`.  (Moving it to the Plugin class makes more logical sense.)

Reply:
With the constants now in the Plugin class, remove the definitions from the Form class, and instead reference them from the Plugin when needed in the Form, e.g. `$this->_plugin::CONSTANT_NAME`.